cluster-autoscaler/core/utils/utils.go

```diff
@@ -181,13 +183,27 @@ func GetNodeInfoFromTemplate(nodeGroup cloudprovider.NodeGroup, daemonsets []*ap
 	}
 	fullNodeInfo := schedulerframework.NewNodeInfo(pods...)
 	fullNodeInfo.SetNode(baseNodeInfo.Node())
-	sanitizedNodeInfo, typedErr := sanitizeNodeInfo(fullNodeInfo, id, ignoredTaints)
+	SetNodeInfoBuiltFromTemplate(fullNodeInfo)
+	sanitizedNodeInfo, typedErr := SanitizeNodeInfo(fullNodeInfo, id, ignoredTaints)
 	if typedErr != nil {
 		return nil, typedErr
 	}
 	return sanitizedNodeInfo, nil
 }
 
+// SetNodeInfoBuiltFromTemplate marks a NodeInfo as generated from a cloud provider's TemplateInfo()
+func SetNodeInfoBuiltFromTemplate(nodeInfo *schedulerframework.NodeInfo) {
+	node := nodeInfo.Node()
+	node.Status.NodeInfo.MachineID = templatedMachineID
```

Comment:
I have mixed feelings about using a semi-random field in node status for this. It's hard to guarantee we'll never have support cloudprovider that use this field for something. Maybe an annotation would be more appropriate? I don't have a strong feeling here though.

Reply:
Thanks, an annotation is much cleaned indeed, just did that!